fakeredis.py

```diff
@@ -326,11 +326,18 @@ def __exit__(self, exc_type, exc_value, traceback):
         self.release()
 
     def acquire(self, blocking=True, blocking_timeout=None):
-        return self.lock.acquire(blocking)
+        owner_id = _decode(self.redis.get(self.name))
+        can_acquire = not owner_id or owner_id == self.id
+        if can_acquire:
+            self.redis.set(self.name, self.id, ex=self.timeout)
+        elif blocking:
+            raise ValueError('fakeredis can\'t do blocking locks')
```

Comment:
I originally had an implementation to sleep and wait for the lock to be released, but eventually changed it to an error because blocking would likely wait forever or until the `blocking_timeout`. In either case I think it's an error and it's better to expose an exception instead of an infinite wait.